### PR TITLE
Post 20: Candid for engineers

### PR DESCRIPTION
**Pull request by roman-kashitsyn**

### CODE REVIEW

**Comment by roman-kashitsyn** [comment]
Closes #39.

File: posts/20-candid-for-engineers.html.pm [hunk 1]
```diff
@@ -0,0 +1,908 @@
...
+
+◊figure{
+
+◊marginnote["mn-rose-tree"]{
+  A definition of a ◊a[#:href "https://en.wikipedia.org/wiki/Rose_tree"]{rose tree} data type containing 32-bit integers (top) and the Candid representation of the same type (bottom).
+}
+
+◊source-code["rust"]{
+◊em{// Rust}
+pub enum Tree { Leaf(i32), Forest(Vec<Tree>) }
```
Comment (chenyan-dfinity) [inline posts/20-candid-for-engineers.html.pm:519]: `Box<Tree>`

File: posts/20-candid-for-engineers.html.pm [hunk 2]
```diff
@@ -0,0 +1,908 @@
...
+
+
+◊subsection-title["faq-remove-alternative"]{Can I add a variant alternative?}
+
+◊p{
+  It depends on the type variance.
+}
+
+◊p{
+  If the variant appears only in method arguments, you can safely add new alternatives.
```
Comment (chenyan-dfinity) [inline posts/20-candid-for-engineers.html.pm:789]: We may also mention the `opt variant` case, where you can safely add and remove variant tags.

**Review by chenyan-dfinity — COMMENTED** [review]
Nice!

File: posts/20-candid-for-engineers.html.pm [hunk 3]
```diff
@@ -0,0 +1,908 @@
...
+◊source-code["candid"]{
+◊b{service} Token : ◊b{(init_balances : vec record { principal; nat })} -> {
+  balance : (record { of : principal }) -> (nat) query;
+  ◊em{// ...}
+}
+}
+}
+
+◊p{
+  Init arguments describe the value that the canister maintainers must specify when they instantiate the canister.
```
[the remark this answers is not in the document]
Comment (chenyan-dfinity) [reply posts/20-candid-for-engineers.html.pm:123]: Also good to mention that the .did file can be fetched from the canister metadata.

**Pull request merged**